src/components/molecules/Visualizer/Plugin/hooks.ts

```diff
@@ -46,6 +57,18 @@ export default function ({
     extended: boolean | undefined,
   ) => void;
 }) {
+  const modalCanBeVisible = useRef<boolean>(false);
+  const popupCanBeVisible = useRef<boolean>(false);
```

Comment:
Perhaps modalVisible and popupVisible are clearer, though I do feel like the logic inside the plugin atom is a little messy

Reply:
Updated.